### PR TITLE
Add test for Llama3.1 ScaledRoPE

### PR DESCRIPTION
**Pull request by joecummings**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

#### Changelog
* Rename `Llama3_1RotaryPositionalEmbeddings` --> `Llama3ScaledRoPE`
* Add test for `Llama3ScaledRoPE`

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help.)

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [x] add unit tests for any new functionality
- [ ] update docstrings for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
	- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (felipemello1) [inline tests/torchtune/models/llama3_1/test_position_embeddings.py:114]: n00b question: The input here is different than in the previous test, but we still expect the same sum() self.EXPECTED_X_OUT_SUM? Should they be different?

Comment (joecummings) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:114]: L117 I believe answers your question

Comment (felipemello1) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:114]: I see, you do broadcasting inside of rope implementation

```
# tensor has shape [b, s, 1, h_d // 2, 2] if packed samples,
# otherwise has shape [1, s, 1, h_d // 2, 2]
rope_cache = rope_cache.view(-1, xshaped.size(1), 1, xshaped.size(3), 2)
```

So the sum should be the same, because the input pos is the same for both:
```
>>> seq_len=3
>>> bsz=2
>>> print(torch.arange(seq_len).unsqueeze(0).expand(bsz, seq_len))
tensor([[0, 1, 2],
        [0, 1, 2]])

>>> print(torch.arange(seq_len))       
tensor([0, 1, 2])
```

Maybe I am missing the point, but I am not sure how this is testing packed samples. Should we try to have a more complex input_pos for this case?, e.g.:
tensor([[0, 1, 2, 0, 1, 2, 3, 4],
        [0, 1, 2, 0, 0, 0 ,0]])

Other than that, LGTM!

Comment (joecummings) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:114]: I definitely agree with your second point and would be in favor of filing an issue for a more complex test! For this, I just wanted to copy-paste from the original test_positional_embeddings and make sure the scaling worked as expected.

Comment (felipemello1) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:114]: Maybe add a TODO before merging, if you think it is worth it

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (ebsmothers) [inline tests/torchtune/models/llama3_1/test_position_embeddings.py:40]: nit: I feel like it's overkill to assert on all of these. Nbd for this PR though

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (ebsmothers) [inline tests/torchtune/models/llama3_1/test_position_embeddings.py:64]: Maybe I'm being dense but why don't we have to do this on every test case?

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (ebsmothers) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:114]: Based on this discussion would also consider renaming from `test_forward_with_packed_pos` as I think it's a bit misleading (maybe `test_forward_with_2d_pos_ids` or something)

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (ebsmothers) [inline tests/torchtune/models/llama3_1/test_position_embeddings.py:137]: Can also explicitly assert that nothing is on meta device

**Review by ebsmothers — APPROVED** [review]
Couple small comments, stamping to unblock

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (joecummings) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:40]: BUT HOW WILL WE KNOW IT'S THE SAME

File: tests/torchtune/models/llama3_1/test_position_embeddings.py
Comment (joecummings) [reply tests/torchtune/models/llama3_1/test_position_embeddings.py:64]: This is the only test that doesn't call the rope forward method, which is where the cache is actually set. So for this one and only this one, we need to manually initialize the cache.

**Pull request merged**
